rust/arrow/src/array.rs

```diff
@@ -249,6 +263,116 @@ impl<T: ArrowNumericType> PrimitiveArray<T> {
     }
 }
 
+impl<T: ArrowTemporalType + ArrowNumericType> PrimitiveArray<T>
+where
+    i64: std::convert::From<T::Native>,
+{
+    /// Returns value as a chrono `NaiveDateTime`, handling time resolution
+    ///
+    /// If a data type cannot be converted to `NaiveDateTime`, a `None` is returned
+    /// TODO: extract constants into static variables
+    pub fn datetime(&self, i: usize) -> Option<NaiveDateTime> {
+        if !self.is_valid(i) {
```

Comment:
I don't think you should check the validity of an array slot.  These methods are similar to `value` so you should allow the caller to check the validity of an array slot themselves.  i.e. the caller may know that there are no nulls and so would not want to check this on every call to this method.

In fact, I would consider changing these method names to `value_as_datetime`, `value_as_date` to make this connection.

Reply:
Thanks, I've updated the names and removed the validity check